feat(telemetry): add configurability of span attributes in logs (#5867)
apollographql/router: .changesets/feat_bnjjj_feat_5540.md

> ### Telemetry: add configurability of span attributes in logs ([Issue #5540](https://github.com/apollographql/router/issues/5540))

Reviewer: ```suggestion
### Add configurability of span attributes in logs ([Issue #5540](https://github.com/apollographql/router/issues/5540))
```

> The `telemetry.exporters.logging.stdout.format.json.span_attributes` option is useful if you don't want to display all spans attributes but only some of them. It takes an array of attribute name to include as a root logging attribute.

Reviewer: ```suggestion
The router supports a new  `telemetry.exporters.logging.stdout.format.json.span_attributes` option that enables you to choose a subset of all span attributes to display in your logs.

When `span_attributes` is specified, the router searches for the first attribute in its input list of span attributes from the root span to the current span and attaches it to the outermost JSON object for the log event. If you set the same attribute name for different spans at different levels, the router chooses the attributes of child spans before the attributes of parent spans.

```

> This will search for the first attribute in the list of span attributes from the root span to the current one and attach it to the outmost json object for the log event.
> If you set the same attribute on different spans at different level, then the identical attribute from child spans will take precedence over the one found previously in parent spans.

Reviewer: ```suggestion

```

> For example, if you have spans that contains `graphql.document` attribute and you only want to display this span attribute:

Author: ```suggestion
For example, if you have spans that contains `span_attr_1` attribute and you only want to display this span attribute:
```

```yaml title="router.yaml"
telemetry:
  exporters:
     logging:
       stdout:
         enabled: true
         format: 
           json:
             display_span_list: false
             span_attributes:

>              - span_attr_1

Reviewer: ```suggestion
             - graphql.document
```

Author: I modified the original sentence to keep using `span_attr_1`

```

Example output with a list of spans:

```json
{
  "timestamp": "2023-10-30T14:09:34.771388Z",
  "level": "INFO",
  "fields": {
    "event_attr_1": "event_attr_1",
    "event_attr_2": "event_attr_2"
  },
  "target": "event_target",
  "span_attr_1": "span_attr_1"
}
```

> 

Reviewer: ```suggestion

To learn more, go to [`span_attributes`](https://www.apollographql.com/docs/router/configuration/telemetry/exporters/logging/stdout#span_attributes) docs.
```
